homeassistant/components/cast/media_player.py

```diff
@@ -534,20 +534,31 @@ async def async_play_media(self, media_type, media_id, **kwargs):
             media_type = sourced_media.mime_type
             media_id = sourced_media.url
 
+        extra = kwargs.get(ATTR_MEDIA_EXTRA, {})
+
         # If media ID is a relative URL, we serve it from HA.
         # Create a signed path.
         if media_id[0] == "/":
+            # Configure play command for when playing a HLS stream
+            if media_id.startswith("/api/hls/"):
+                extra = {
+                    **extra,
+                    "stream_type": "LIVE",
+                    "media_info": {
+                        "hlsVideoSegmentFormat": "fmp4",
+                    },
+                }
+
             media_id = async_sign_path(
                 self.hass,
                 quote(media_id),
                 timedelta(seconds=media_source.DEFAULT_EXPIRY_TIME),
             )
 
             # prepend external URL
-            hass_url = get_url(self.hass, prefer_external=True)
+            hass_url = get_url(self.hass, prefer_external=self.hass.config.api.use_ssl)
```

Comment:
If a user has HTTPS configured locally, Cast can never resolve it.

Previously camera.play_stream would always send a local URL. This is wrong.